src/composites/TreeDisplay/nodeDescriptionPanel/NodeDescriptionPanel.tsx

```diff
@@ -2,34 +2,75 @@ import "./NodeDescriptionPanel.css"
 import {useEffect, useState} from "react";
 import {determineNodeInfo} from "./NodeDescriptionPanelHelpers";
 
-export default function NodeDescriptionPanel(props) {
+export default function NodeDescriptionPanel(props: { nodes: any[]; impacts: any }) {
+  const [nodes, setNodes] = useState<any[]>([]);
+  const [orderBy, setOrderBy] = useState<string>("default");
+  const [orderDirection, setOrderDirection] = useState<string>("asc");
 
-    const [nodes,setNodes] = useState([]);
+  // Rerender this component whenever the nodes prop from TreeDisplay changes.
+  useEffect(() => {
+    setNodes(props.nodes);
+  }, [props.nodes]);
 
-    // Rerender this component whenever the nodes prop from TreeDisplay changes.
-    useEffect(() => {
-        setNodes(props.nodes)
-    },[props.nodes])
+  function makeNodePanelRectangles() {
+    let orderedNodes = [...nodes];
 
-    function determinePanelHeight() {
-        const height = 100/nodes.length;
-        if (height > 34) return 33.33;
-        return height;
+    switch (orderBy) {
+      case "alphabetical":
+        orderedNodes.sort((a, b) => (orderDirection === "asc" ? a.name.localeCompare(b.name) : b.name.localeCompare(a.name)));
+        break;
+      case "nodeType":
+        orderedNodes.sort((a, b) => (a.node_type > b.node_type ? 1 : -1));
+        break;
+      case "value":
+        orderedNodes.sort((a, b) => (orderDirection === "asc" ? a.value - b.value : b.value - a.value));
+        break;
+      default:
+        break;
     }
 
-    function makeNodePanelRectangles() {
-        return (nodes.map( (node,i = node.hashCode()) => (
-                <div className={`${i === nodes.length-1 ? (nodes.length > 2 ? "node-bottom-panel" : "node-panel") : "node-panel"}`} style={{height : `${determinePanelHeight()}%`}} key={i}>
-                    {determineNodeInfo(node,props.impacts)}
-                </div>
-                )
-            )
-        )
-    }
+    return orderedNodes.map((node, i) => (
+      <div
+        className={`${
+          i === nodes.length - 1 ? (nodes.length > 2 ? "node-bottom-panel" : "node-panel") : "node-panel"
+        }`}
+        key={i}
+      >
+        {determineNodeInfo(node, props.impacts)}
+      </div>
+    ));
+  }
+
+  const handleOrderByChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
```

Comment:
We should only set orderby here, and set a default state for setOrderDirection. If a user already had set order direction to descending, and they change the order by key, then this would reset the order direction as well, which is undesirable.

Reply:
handleOrderByChange doesn't call setOrderDirection anymore.